tensorflow/core/kernels/maxpooling_op_gpu.cu.cc

```diff
@@ -28,6 +28,21 @@ limitations under the License.
 
 namespace tensorflow {
 namespace {
+// Checks env var to determine whether maxpool functions should propagate NaN
+// values. NaNs are propagated unless TF_ENABLE_MAXPOOL_NANPROP=0.
+bool QueryMaxPoolNanPropIsEnabled() {
+  bool default_value = true;
```

Comment:
Check the default behavior is a high risk. You can potentially break models that are already working. 

It is safe to add a flag that works both way, and have the default do the old way. We can have a separate CL trying to flip the default. The default is to announce it, have people try it for a few weeks. If nobody complains, we will then switch. We need to be a lot more careful changing default behavior for existing ops, than to add new ops.